gateleen-routing/src/main/java/org/swisspush/gateleen/routing/DeferCloseHttpClient.java

```diff
@@ -39,54 +34,13 @@ public DeferCloseHttpClient(Vertx vertx, HttpClient delegate) {
         this.delegate = delegate;
     }
 
+
     @Override
-    public HttpClientRequest request(HttpMethod method, int port, String host, String requestURI, Handler<HttpClientResponse> responseHandler) {
+    public void request(HttpMethod method, int port, String host, String requestURI, Handler<AsyncResult<HttpClientRequest>> handler) {
         logger.debug("({}:{}).request({}, \"{}\")", host, port, method, requestURI);
         countOfRequestsInProgress += 1;
         logger.debug("Pending request count: {}", countOfRequestsInProgress);
-
-        // Delegate to the same method on the delegate. But install our own handler which
-        // allows us to intercept the response.
-        HttpClientRequest request = delegate.request(method, port, host, requestURI, upstreamRsp -> {
```

Comment:
@hiddenalpha do we need to migrate this? Or do you think that this is no longer needed with the new vert.x version?

Reply:
Removing this defeats the purpose of the whole class (*aka breaks it*). I bet we still need this. Because without the reference counting, we have the old behavior if the class wouldn't exist at all. And therefore we will re-introduce the bug we solved with it.

Reply:
Kind of expected that we probably continue to need it. Would you be OK if we merge now to develop
- with this open issue
- and consider the release not fit for production
- and follow up right away
?

The reason I ask is this: I would like to integrate the release in our project and run all automated integration tests. In order to do this we need to stage to maven central (which means to release, we normally do this through develop / master). Technically I can also release from this branch I believe. An other advantage to have it arrive in develop soon  (beside the ability to release as usual) is to avoid future merge conflicts.

Reply:
Discussed, we add a TODO, maybe something like

TODO: Migrate the original implementation

and then also look into it right away. An option is probably also to introduce this DeferCloseHttpClient in vert.x itself.